Runtime/Scripts/Extensions/MaterialExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+
+namespace Unity.MeshSync {
+public static class MaterialExtensions {
+    /// <summary>
+    /// Sets the given texture on the material.
+    /// If there was a render texture in the slot, it is released.
+    /// </summary>
+    /// <param name="mat">The material to set the texture on</param>
+    /// <param name="name">The name of the texture slot</param>
+    /// <param name="texture">The texture to set</param>
+    public  static void SetTextureSafe(this Material mat, string name, Texture texture) {
+        // If there was a renderTexture set that we're clearing now, make sure to release its resources:
+        var existingTexture = mat.GetTexture(name);
+        if (existingTexture is RenderTexture existingRenderTexture &&
+            texture != existingRenderTexture) {
+            existingRenderTexture.Release();
```

Comment:
This is a helper method that ensures that render textures that are replaced with other textures are released.
The class that created the render texture does not know when another texture is set on the material to replace the texture. The caller has no reference to the render texture and should not have to care what texture was previously set.

Reply:
That approach assumes that we always need to release RenderTextures, which may not be true in general case.

For example, when the materials have pre-created RenderTextures, instead of the ones created automatically by the code in this PR.

In the future, someone may use this `SetTextureSafe()` function to replace a texture of a material, and find his/her RenderTextures destroyed even though that's not intended.

Reply:
Not sure how else we can handle this. This method handles our specific scenario where any render textures that were previously set need to be released to avoid memory leaks. If someone wants to hold on to the render texture they can call the other SetTexture method.

Reply:
Okay, at least we should rename the function then.

`SetTextureAndReleaseExisting` perhaps ?

Reply:
Done